aria_data_loaders/aria_data_utils/spot_aria_p1_node.py

```diff
@@ -0,0 +1,591 @@
+            ).parent_tform_child
+        )
+        hand_mn_wrist_T_handcam = self.spot.convert_transformation_from_BD_to_magnum(
+            hand_bd_wrist_T_handcam_dict
+        )
+
+        hand_bd_body_T_wrist_dict = (
+            frame_tree_snapshot_hand.child_to_parent_edge_map.get(
+                "arm0.link_wr1"
+            ).parent_tform_child
+        )
+        hand_mn_body_T_wrist = self.spot.convert_transformation_from_BD_to_magnum(
+            hand_bd_body_T_wrist_dict
+        )
+
+        hand_mn_body_T_handcam = hand_mn_body_T_wrist @ hand_mn_wrist_T_handcam
+
+        return hand_mn_body_T_handcam
+
+    def get_spot_a_T_b(self, a: str, b: str) -> sp.SE3:
+        frame_tree_snapshot = (
+            self.spot.get_robot_state().kinematic_state.transforms_snapshot
+        )
+        se3_pose = get_a_tform_b(frame_tree_snapshot, a, b)
+        pos = se3_pose.get_translation()
+        quat = se3_pose.rotation.normalize()
+        return sp.SE3(quat.to_matrix(), pos)
+
+    def _get_body_T_headcam(self, frame_tree_snapshot_head):
+        raise RuntimeError("This method is not used anymore. Please use get_spot_a_T_b")
+
+    def _get_spotWorld_T_handcam(
+        self, frame_tree_snapshot_hand, spot_frame: str = "vision"
+    ):
+        if spot_frame != "vision" and spot_frame != "odom":
+            raise ValueError("spot_frame should be either vision or odom")
+        spot_world_frame = spot_frame
+
+        hand_bd_wrist_T_handcam_dict = (
+            frame_tree_snapshot_hand.child_to_parent_edge_map.get(
+                "hand_color_image_sensor"
+            ).parent_tform_child
+        )
+        hand_mn_wrist_T_handcam = self.spot.convert_transformation_from_BD_to_magnum(
+            hand_bd_wrist_T_handcam_dict
+        )
+
+        hand_bd_body_T_wrist_dict = (
+            frame_tree_snapshot_hand.child_to_parent_edge_map.get(
+                "arm0.link_wr1"
+            ).parent_tform_child
+        )
+        hand_mn_body_T_wrist = self.spot.convert_transformation_from_BD_to_magnum(
+            hand_bd_body_T_wrist_dict
+        )
+
+        hand_bd_body_T_spotWorld_dict = (
+            frame_tree_snapshot_hand.child_to_parent_edge_map.get(
+                spot_world_frame
+            ).parent_tform_child
+        )
+        hand_mn_body_T_spotWorld = self.spot.convert_transformation_from_BD_to_magnum(
+            hand_bd_body_T_spotWorld_dict
+        )
+        hand_mn_spotWorld_T_body = hand_mn_body_T_spotWorld.inverted()
+
+        hand_mn_body_T_handcam = hand_mn_body_T_wrist @ hand_mn_wrist_T_handcam
+        hand_mn_spotWorld_T_handcam = hand_mn_spotWorld_T_body @ hand_mn_body_T_handcam
+
+        return hand_mn_spotWorld_T_handcam
+
+    def _get_spotWorld_T_headcam(
+        self, frame_tree_snapshot_head, use_vision_as_world: bool = True
+    ):
+        raise RuntimeError("This method is not used anymore. Please use get_spot_a_T_b")
+
+    def get_avg_spotWorld_T_marker_HAND(
```

Comment:
Duplication of averaging code. Let's turn this into a utility and use in a generic manner.

Reply:
We have a generic function for averaging. This will be taken care of in the PR to the issue - https://github.com/facebookresearch/spot-sim2real/issues/90